### ISSUE TITLE
Threshold for noisy points

### ISSUE THREAD

**Issue by rth** [scikit-learn-contrib/hdbscan#77]
I was wondering if  there is a way to control how many points will be marked as "noisy" in HBSCAN? For instance, in [`DBSCAN`](http://scikit-learn.org/stable/modules/generated/sklearn.cluster.DBSCAN.html) there is an `eps` parameter that affects the cluster structure, but also effectively how many points will be marked as noisy. 

Let's consider the following example (which doesn't have clearly defined dense clusters),
```py
import numpy as np
import matplotlib.pyplot as plt
from sklearn.manifold import TSNE
from hdbscan import HDBSCAN

np.random.seed(3)
X = np.random.rand(100, 20)
projection = TSNE().fit_transform(X)

plt.scatter(*projection.T)

model = HDBSCAN(min_cluster_size=2, )
model.fit(X)
cm = plt.scatter(*projection.T, c=model.labels_, cmap='brg')
cb = plt.colorbar(cm)
cb.set_label('Cluster label')
mask_noisy = (model.labels_ == -1)
print('Clusters found:', len(np.unique(model.labels_[~mask_noisy])))
print('Noisy points: {} / {}'.format(mask_noisy.sum(), len(mask_noisy)))
```
which returns
```
Clusters found: 2
Noisy points: 58 / 100
```
![test](https://cloud.githubusercontent.com/assets/630936/21188628/42a9be34-c21c-11e6-96ce-e192b3b73c85.png)

most points will be marked as noisy, and if we look at the outliers scores, `model.outlier_scores_.max() == 0.19`. 
  1. Is there a way to indicate to HBSCAN that a point should not be considered noisy (e.g. based on some `outlier_scores_` threshold)?
  2. what would be the easiest/most efficient way to merge a noisy point back to the nearest cluster? Would it be to walk though the nodes of the minimum spanning tree, until a cluster is encountered?
  3. I was also wondering how noisy points are handled in the [cluster hierarchy](https://hdbscan.readthedocs.io/en/latest/how_hdbscan_works.html#build-the-cluster-hierarchy) are they considered as a separate cluster or just ignored?

Thanks a lot!
P.S: also all the classes from `hdbscan/plots.py` such as `SingleLinkageTree` etc appear to be missing from the [API documentation](https://hdbscan.readthedocs.io/en/latest/api.html)...

**Edit:** OK, I just realized that this might be a duplicate of issue #72

**Comment by lmcinnes** [comment]
The parameter you are looking for is min_samples (which works equivalently
to DBSCAN). If that isn't fine tuned enough you can also play with alpha,
although that requires a little more care and I don't advise it unless you
need to change it. You can see more in the documentation:
https://hdbscan.readthedocs.io/en/latest/parameter_selection.html

**Comment by rth** [comment]
@lmcinnes Thanks! I did confuse `min_cluster_size` with `min_samples`. With the above example,  decreasing `min_samples` up to 2 doesn't change anything, while setting `min_samples=1` yields 25 clusters with 33 / 100 noisy points. So it does reduce the amount of noisy labels, but only up to a point.

Closing this issue as duplicate of https://github.com/scikit-learn-contrib/hdbscan/issues/72, where you mentioned, 
> I am currently working on code that can provide a membership vector, proving the probability that a given point is in each of the found clusters. This is currently in the prediction branch and not complete yet. It might satisfy your desire to assign everything.

which would solve my problem as well.
